File: Java/187. Repeated DNA Sequences.java
package Java;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;

class Solution {
    public List<String> findRepeatedDnaSequences(String s) {
        List<String> ans = new ArrayList<String>();
        if (s.length() <= 10) return ans;
        HashMap<String, Integer> mymap = new HashMap<String, Integer>();
        for (int i = 10; i <= s.length(); i++) {
            String subString = s.substring(i - 10, i);
            if (!mymap.containsKey(subString)) mymap.put(subString, 1);
            else if (mymap.get(subString) == 1){
                ans.add(subString);
                mymap.put(subString, mymap.get(subString) + 1);
            }
        }
        return ans;
    }
}

File: Python/187. Repeated DNA Sequences.py
class Solution(object):
    def findRepeatedDnaSequences(self, s):
        """
        :type s: str
        :rtype: List[str]
        """
        if len(s) <= 10:
            return []
        mymap = {}
        ans = []
        for i in range(len(s) - 10 + 1):
            subString = s[i: i + 10]
            if mymap.get(subString) == None:
                mymap[subString] = 1
            elif mymap[subString] == 1:
                ans.append(subString)
                mymap[subString] += 1
        return ans